### ISSUE TITLE
[Bug] False positive of CA1508 triggered in static classes

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn-analyzers#5398]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug

False positive for CA1508.

Only triggered in static classes.

### Steps To Reproduce
````cs
namespace CA1508 {
  public static class CA1508 {


	// return object or null (50/50 odds)
    public static object? Create() => (new Random().Next(0, 2) != 0) ? new object() : null;   


    public static void Bug() {
      var obj = Create();
      if (obj == null) throw new InvalidOperationException();   // <------------ CA1508 here
      obj.ToString();
    }


  }
}
````

Warning:
> 'obj == null' is always 'false'. Remove or refactor the condition(s) to avoid dead code. csharp(CA1508)

### Expected behavior
Warning should not appear

### Actual behavior
Warning appears.

If I remove `static` then warning disappears. Bug has something to do with static classes.

## Additional context
n/a

**Comment by Youssef1313** [comment]
This doesn't reproduce for me if the `Create` method's return type is null-annotated. But reproduces in null-oblivious contexts, which is intended:

https://github.com/dotnet/roslyn-analyzers/blob/60eb13c44c366205e5a4dec195404477009c41de/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs#L936

**Comment by mavasani** [comment]
Thank you for your analysis @Youssef1313. Going to close this as by-design.
